opensrp-opd/src/main/java/org/smartregister/opd/utils/OpdUtils.java

```diff
@@ -323,4 +325,31 @@ public static int buildRepeatingGroupTests(@NonNull JSONObject step1JsonObject)
         return compositeObsArrayList;
     }
 
+    public static void injectRelevanceFields(@NonNull JSONObject jsonForm, @NonNull Map<String, String> clientDetailsMap) {
```

Comment:
Method name does not do what is says. Can we have this code reusable at least. I can see data duplication

Reply:
Suggestions on the naming?...Its in utils, how reusable do you want it?

Reply:
I can see that you are repeating  the `type` & `hidden`. With the key & value static. Ådding an additional field to be injected requires ~9 more lines to do the same thing. The only change being the values in `value` and `key`? What do you think?

Reply:
So am thinking i add the fields in the json form then update their values....because what if type changes also or a particular field is needed

Reply:
Do we need a different type of widget for injected fields if the values are already in the DB and are strings? 🤷‍♂  These  relevance fields are reference fields only right?

Reply:
hehe.....its a good idea but do you want to work on it?

Reply:
Cool, let me put it this way. That needs to be added for this PR in this PR. If you are going to go around deleting code and adding your own code, it'd rather add some value, else I don't know why the refactor is needed

Reply:
Cool we can table a discussion for this.